client/allocrunner/groupservice_hook.go

```diff
@@ -117,10 +127,33 @@ func (h *groupServiceHook) Update(req *interfaces.RunnerUpdateRequest) error {
 	return h.consulClient.UpdateWorkload(oldWorkloadServices, newWorkloadServices)
 }
 
-func (h *groupServiceHook) Postrun() error {
+func (h *groupServiceHook) RunnerPreKillHook() {
 	h.mu.Lock()
 	defer h.mu.Unlock()
+
+	if h.delay == 0 {
```

Comment:
I think we should just get rid of this statement. Always deregistering services before killing the TRs seems like the right thing to do.

Reply:
So then just get rid of the postrun hook for group service? https://github.com/hashicorp/nomad/blob/master/client/allocrunner/groupservice_hook.go#L120

Reply:
I think we still need the postrun hook to deregister services when the alloc exits normally.

Reply:
Ah right, is the call idempotent or should we keep the check to see if we have already de-registered?